compiler/rustc_resolve/src/ident.rs

```diff
@@ -296,7 +296,6 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
         ribs: &[Rib<'ra>],
         ignore_binding: Option<NameBinding<'ra>>,
     ) -> Option<LexicalScopeBinding<'ra>> {
-        assert!(ns == TypeNS || ns == ValueNS);
```

Comment:
Are there tests for various situations with naming conflicts?

Reply:
hm, unsure, In theory there shouldn't be any issue, EII attributes are simply part of the macro namespace and act exactly as other macros. The reason this assertion had to be removed is that we use the defid of that macro as an identifier for the EII itself - with is we look up its signature when type checking for example. And we used to effectively assert that all defids of macros were filtered out at this point.